lib/endpoints/Queue.js

```diff
@@ -199,20 +201,31 @@ class Queue {
     }
 }
 
-const queueMap = new Map();
+/**
+ * Keeps track of all the queues.
+ */
+class QueueScheduler {
```

Comment:
ok, so, I've taken the time to read through this now, and it looks good, I do like that it will delete an empty queue.  I'm curious why the change to a class for the map of queues, though.  Since it is only used once, I had made it module level, so that it would be a singleton type of construct.  Is it just for the test (thanks for adding that!), or is there an intent to maybe have multiple instances of it?

Reply:
There are two reasons why I did that:
1) Readability: the functions are all related, by bundling the functions in one class everyone reading the code knows that these functions are related. 
2) Future proofing: We might want the class to control the queues.